runtime/startup_byt.c

```diff
@@ -450,8 +450,11 @@ CAMLexport void caml_main(char_os **argv)
   caml_close_channel(chan); /* this also closes fd */
   caml_stat_free(trail.section);
   /* Ensure that the globals are in the major heap. */
-  caml_oldify_one (caml_global_data, &caml_global_data);
-  caml_oldify_mopup ();
+  {
+    CAMLparam1(caml_global_data);
+    caml_minor_collection();
```

Comment:
By manually calling `caml_oldify_mopup` without `caml_empty_minor_heap`, this left the minor heap in a weird state. The weirdness was confined to dead data, so this didn't crash. However, the new debug-mode checks scan even the dead minor data for well-formedness, which triggered an assertion failure here. The change is to do a real minor GC instead of a manual half-GC.

(You're right that the change is missing from the other startup path)